src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubStreamOptions.cs

```diff
@@ -23,6 +23,15 @@ public class EventHubOptions
         /// Hub path.
         /// </summary>
         public string Path { get; set; }
+        /// <summary>
+        /// The token credential.
+        /// </summary>
+        public TokenCredential TokenCredential { get; set; }
```

Comment:
If this is a secret, we should add a `[Redact]' attribute so it is not logged at silo startup

Reply:
Should not be a secret. This is the runtime account/principal used. It has a `GetTokenAsync()` method that then gets the access_token for the resources.